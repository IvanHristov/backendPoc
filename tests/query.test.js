const gql = require('graphql-tag')
const createTestServer = require('./helper')

const user = { id: 1, first_name: 'ivan', last_name: 'hristov', email: 'ih95@gmail.com' }
const CONTACTS = gql`
  {
    contacts {
      id
      first_name
      last_name
      email
    }
  }
`
const CONTACT = gql`
  {
    contact(input: {id: 5}) {
      id
      first_name
      last_name
      email
    }
  }
`

describe('queries', () => {
  test('contacts', async () => {
    const { query } = createTestServer({
      models: {
        Contacts: {
          findMany: jest.fn(() => [user])
        }
      }
    })

    const res = await query({ query: CONTACTS })
    expect(res).toMatchSnapshot()
  }),
    test('contact ', async () => {
      const { query } = createTestServer({
        models: {
          Contacts: {
            findOne: jest.fn(() => user)
          }
        }
      })

      const res = await query({ query: CONTACT })
      expect(res).toMatchSnapshot()
    });
})
