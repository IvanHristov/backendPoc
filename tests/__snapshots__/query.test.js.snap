// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`queries contact  1`] = `
Object {
  "data": Object {
    "contact": Object {
      "email": "ih95@gmail.com",
      "first_name": "ivan",
      "id": "1",
      "last_name": "hristov",
    },
  },
  "errors": undefined,
  "extensions": undefined,
  "http": Object {
    "headers": Headers {
      Symbol(map): Object {},
    },
  },
}
`;

exports[`queries contacts 1`] = `
Object {
  "data": Object {
    "contacts": Array [
      Object {
        "email": "ih95@gmail.com",
        "first_name": "ivan",
        "id": "1",
        "last_name": "hristov",
      },
    ],
  },
  "errors": undefined,
  "extensions": undefined,
  "http": Object {
    "headers": Headers {
      Symbol(map): Object {},
    },
  },
}
`;
